app/serializers/device_serializer.rb

```diff
@@ -0,0 +1,7 @@
+class DeviceSerializer < ActiveModel::Serializer
+  attributes *%i(id location location_name payload created_at updated_at)
+
+  def location
```

Comment:
Not sure if device needs a location or not. I'm sure it will be useful to have a notion of general area (like you have with `location_name`), but since solarcast is a semi-mobile device, it's "location" will likely be wherever it most recently reported a measurement.